trace-forward/src/Trace/Forward/Acceptor.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE LambdaCase #-}
+
+-- | This top-level module will be used by the acceptor application.
+--   Acceptor application asks 'TraceObject's from the forwarder application.
+module Trace.Forward.Acceptor
+  ( runTraceAcceptor
+  ) where
+
+import qualified Codec.Serialise as CBOR
+import           Control.Concurrent.STM.TBQueue (TBQueue)
+import           Control.Exception.Extra (try_)
+import           Data.Typeable (Typeable)
+import           System.Time.Extra (sleep)
+
+import           Ouroboros.Network.Util.ShowProxy (ShowProxy(..))
+
+import           Trace.Forward.Network.Acceptor (listenToForwarder)
+import           Trace.Forward.Configuration (AcceptorConfiguration (..))
+import           Trace.Forward.Protocol.Type (NodeInfoStore)
+
+runTraceAcceptor
+  :: (CBOR.Serialise lo,
+      ShowProxy lo,
+      Typeable lo)
+  => AcceptorConfiguration lo -- ^ Acceptor configuration.
+  -> TBQueue lo               -- ^ All 'TraceObject's received from the node will be written in this queue.
```

Comment:
This means a pull based protocol is translated into a push based application, though the boundedness of the queue might help a bit with back-pressure. Is this necessary tough?  Why cannot we process the data in the protocol handlers?

Reply:
@coot Sorry, I didn't understand... All `TraceObject`s received from the forwarder should be stored somewhere, right? We think that `TBQueue` is a good solution.

Reply:
You don't need to store them, you just need to process them and if you do that inside protocols handlers (without a queue) you will get better back-pressure, i.e. if the application is slow to process the data it will slow down the pace of requesting more data.

Reply:
Ok, I removed the queue on the acceptor's side, now the handler is provided.